File: KnowledgeSharingVue/knowledge-sharing-vue/src/js/models/views/view-user-conversation.js
// Technology Stack: C# 

import UserConversation from '../entities/user-conversation';

class ViewUserProfileConversation extends UserConversation {
    constructor() {
        super();
        this.Username = null;
        this.FullName = null;
        this.Avatar = null;
        this.Cover = null;
    }

    init() {
        return new ViewUserProfileConversation();
    }
}

export default ViewUserProfileConversation;

File: KnowledgeSharingApi/KnowledgeSharingApi.Domain/Models/Entities/Views/ViewUserConversation.cs
﻿using KnowledgeSharingApi.Domains.Models.Entities.Entities;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace KnowledgeSharingApi.Domains.Models.Entities.Views
{
    [Table("ViewUserConversation")]
    public class ViewUserConversation : UserConversation
    {
        public string Username { get; set; } = string.Empty;

        public string FullName { get; set; } = string.Empty;

        public string? Avatar {  get; set; }

        public string? Cover { get; set; }



        protected override ViewUserConversation Init()
        {
            return new();
        }
    }
}
